session/pingpong/hermes_promise_handler.go

```diff
@@ -178,6 +177,13 @@ func (aph *HermesPromiseHandler) handleNodeStopEvents(e event.Payload) {
 func (aph *HermesPromiseHandler) requestPromise(er enqueuedRequest) {
 	defer func() { close(er.errChan) }()
 
+	providerID := er.providerID
+	hermesID := common.HexToAddress(er.em.HermesID)
+	channelID, err := crypto.GenerateProviderChannelID(providerID.Address, hermesID.Hex())
+	if err != nil {
+		er.errChan <- fmt.Errorf("could not generate provider channel address: %w", err)
+	}
```

Comment:
you should probably return here, no point in continuing if you don't have a channel id

Reply:
nice catch, fixed